### DISCUSSION TITLE
Please add an option to set a static IP address

### DISCUSSION THREAD

**Discussion by akeck** [Guilouz/Creality-Helper-Script-Wiki#700] [Features Request]
Hi,

it would be nice if you could add an option to set a static IP address for the printer.
DHCP is a pain when accessing the printer from a home assistant panel and/or Orca Slicer..

best regards :)

**Comment by pauloeduardogodoy** [comment]
I set the static IP using my router. The only pain in the a$$ is the MAC address from the ethernet adapter that changes every firmware reset/update.

**Reply by akeck** [reply]
one of the (many) reasons why I don't like DHCP...

**Comment by CyberCowboy** [comment]
just throwing another voice saying I'd really appreciate this option.   Even if it weren't in the script but a way to manual set it that doesn't get borked with new firmware reset/update.

**Reply by VashBaldeus** [reply]
> just throwing another voice saying I'd really appreciate this option. Even if it weren't in the script but a way to manual set it that doesn't get borked with new firmware reset/update.

Considering how the updates by Creality work, that if we do not restore to factory before we update we face errors (i.e. 1.3.3.36 update). Honestly they took Klipper and botched how it operates with their hardware making it more complex than it has to be. Good example is fan pinout configs and what not. Would've been nice if they at the least consulted Klipper dev team, also in the beginning of the K1 series the firmware was closed-source which was violating the open-source rules.

Wonder if at some point we'll see a board that could be a drop in replacement that we can run Klipper on, cause each time I print something the hardware is pegged on my machine (K1 Max) while the Raspberry Pi 3B+ on my Ender3v2 doesn't even reach 20% utilization while printing. They placed somewhat underpowered hardware there for no apparent reason, Considering the price tag of the machine.
